### PR TITLE
28

### PR DESCRIPTION
**Pull request by AntipovJN**

### CODE REVIEW

File: src/main/java/dao/impl/JDBC/BasketDaoJDBC.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+public class BasketDaoJDBC implements BasketDao {
+
+    private Logger logger = Logger.getLogger(OrderDaoJDBC.class);
+
+    @Override
+    public void addBasket(User user) {
+        try (Connection connection = ConnectionJDBC.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(
+                    "INSERT INTO baskets (user_id) VALUES (?)");
```
Comment (Sofasmile) [inline src/main/java/dao/impl/JDBC/BasketDaoJDBC.java:26]: better move queries from method to class

File: src/main/java/dao/impl/JDBC/OrderDaoJDBC.java [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+public class OrderDaoJDBC implements OrderDao {
+
+    private Logger logger = Logger.getLogger(OrderDaoJDBC.class);
+
+    @Override
+    public void addOrder(Order order) {
+        try (Connection connection = ConnectionJDBC.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(
+                    "INSERT INTO order_test (address, payment, code, products, userid) "
```
Comment (Sofasmile) [inline src/main/java/dao/impl/JDBC/OrderDaoJDBC.java:26]: "INSERT INTO order_test (address, payment, code, products, user_id) "

File: src/main/java/dao/impl/JDBC/OrderDaoJDBC.java [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+            PreparedStatement statement = connection.prepareStatement(
+                    "SELECT order_test.id, address, payment, code, products, userid, email, password, role "
+                            + "FROM order_test INNER JOIN users_test "
+                            + "ON userid = users_test.id WHERE code=? AND userid=?");
+            statement.setString(1, Integer.toString(code.getCodeValue()));
+            statement.setLong(2, code.getUser().getId());
+            ResultSet resultSet = statement.executeQuery();
+            return getOrderFromResultset(resultSet);
+        } catch (SQLException e) {
+            logger.error(String.format("Failed getting order where userid='%s'",
```
Comment (Sofasmile) [inline src/main/java/dao/impl/JDBC/OrderDaoJDBC.java]: if we want to use integer we can write:
"Failed getting order where userid = %d"

File: src/main/java/dao/impl/JDBC/OrderDaoJDBC.java [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+                        new Code(Integer.valueOf(resultSet.getString("code")),
+                                new User(resultSet.getLong("userid"),
+                                        resultSet.getString("email"),
+                                        resultSet.getString("password"),
+                                        resultSet.getString("role"))));
+                order.setId(resultSet.getLong("id"));
+                return Optional.of(order);
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
```
Comment (Sofasmile) [inline src/main/java/dao/impl/JDBC/OrderDaoJDBC.java]: ???

File: src/main/java/utils/SHA256StringHashUtil.java [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    }
+
+    private static String bytesToHex(byte[] bytes) {
+        StringBuffer result = new StringBuffer();
+        for (byte b : bytes)
+            result.append(Integer.toString((b & 0xff) + 0x100, 16).substring(1));
+        return result.toString();
+    }
+
+    public static void main(String[] args) {
```
Comment (Sofasmile) [inline src/main/java/utils/SHA256StringHashUtil.java]: do we need this main method here?

File: src/main/java/utils/SHA256StringHashUtil.java [hunk 6]
```diff
@@ -0,0 +1,28 @@
+package utils;
+
+import java.security.MessageDigest;
+
+public class SHA256StringHashUtil {
+
+    public static String getSha256(String value) {
+        try {
+            MessageDigest md = MessageDigest.getInstance("SHA-256");
```
Comment (Sofasmile) [inline src/main/java/utils/SHA256StringHashUtil.java]: md?

File: src/main/java/utils/SHA256StringHashUtil.java [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+            md.update(value.getBytes());
+            return bytesToHex(md.digest());
+        } catch (Exception ex) {
+            throw new RuntimeException(ex);
```
Comment (Sofasmile) [inline src/main/java/utils/SHA256StringHashUtil.java:18]: we cam use logger here too

File: src/main/java/dao/impl/JDBC/BasketDaoJDBC.java (hunk 1 above)
Comment (AntipovJN) [reply src/main/java/dao/impl/JDBC/BasketDaoJDBC.java:26]: Is it neccessary ? I think it will be not handy..

Comment (Sofasmile) [reply src/main/java/dao/impl/JDBC/BasketDaoJDBC.java:26]: maybe, but it look like more understandable when query is placed separately from the PreparedStatement
and can be a situation when we use one query in two methods

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**